### PR TITLE
Keep Assignment If Squawking Reserved Code

### PR DESCRIPTION
**Pull request by AndyTWF**

### CODE REVIEW

File: app/Listeners/Squawk/ReserveInFirProximity.php [hunk 1]
```diff
@@ -55,12 +56,18 @@ public function handle(NetworkAircraftUpdatedEvent $event): bool
...
+             * - Squawking its assigned code,
+             * - Squawking one of the reserved codes, so probably hasn't set its code yet
+             * - Only recently changed code, so let it go
+             */
             if (
                 (($currentAssignment = $this->squawkService->getAssignedSquawk(
                     $aircraft->callsign
                 )) && $aircraft->transponder == $currentAssignment->getCode()) ||
-                $aircraft->transponder_last_updated_at > Carbon::now()->subMinutes(self::CONSISTENT_SQUAWK_MINUTES)
+                $aircraft->transponder_last_updated_at > Carbon::now()->subMinutes(self::CONSISTENT_SQUAWK_MINUTES) ||
```
Comment (atoff) [inline app/Listeners/Squawk/ReserveInFirProximity.php:69]: I would be tempted to update to something like below for readability but up to you! Fits in well with the comment imo
```
$squawkingAssignedCode = (($currentAssignment = $this->squawkService->getAssignedSquawk(
                    $aircraft->callsign
                )) && $aircraft->transponder == $currentAssignment->getCode());

$squawkingNonAssignableCode = NonAssignableSquawkCode::where('code', $aircraft->transponder)->exists()

$recentlyChangedCode = $aircraft->transponder_last_updated_at > Carbon::now()->subMinutes(self::CONSISTENT_SQUAWK_MINUTES)

if ($squawkingAssignedCode || $squawkingNonAssignableCode ....
```

File: app/Listeners/Squawk/ReserveInFirProximity.php [hunk 2]
```diff
@@ -55,12 +56,18 @@ public function handle(NetworkAircraftUpdatedEvent $event): bool
...
+                NonAssignableSquawkCode::all()->contains($aircraft->transponder)
```
Comment (atoff) [inline app/Listeners/Squawk/ReserveInFirProximity.php:70]: Might be safer (and more efficient?) to use `NonAssignableSquawkCode::where('code', $aircraft->transponder)->exists()` again

File: app/Listeners/Squawk/ReserveInFirProximity.php (hunk 1 above)
Comment (AndyTWF) [reply app/Listeners/Squawk/ReserveInFirProximity.php]: Have refactored into clearer methods :)

File: app/Listeners/Squawk/ReserveInFirProximity.php (hunk 2 above)
Comment (AndyTWF) [reply app/Listeners/Squawk/ReserveInFirProximity.php]: Refactored :)

**Pull request merged**